File: webservinho_app/php-bin/echo.php
#!/usr/bin/php-cgi
<?php
header("Content-Type: text/html");
echo "<h2>PHP CGI Echo</h2>";

if (!empty($_GET)) {
    echo "<p><b>GET Parameters:</b></p><ul>";
    foreach ($_GET as $k => $v) echo "<li>$k = $v</li>";
    echo "</ul>";
}

if (!empty($_POST)) {
    echo "<p><b>POST Parameters:</b></p><ul>";
    foreach ($_POST as $k => $v) echo "<li>$k = $v</li>";
    echo "</ul>";
}

echo "<hr><p><a href='/'>Back to Home</a></p>";
?>

File: webservinho_app/cgi-bin/echo.py
#!/usr/bin/python3
import os, cgi

print("Content-Type: text/html\r\n\r\n")
print("<html><body><h2>Python CGI Echo</h2>")
print("<p><b>Query string:</b> {}</p>".format(os.environ.get("QUERY_STRING", "")))

form = cgi.FieldStorage()
if form:
    print("<ul>")
    for key in form.keys():
        print(f"<li>{key}: {form.getvalue(key)}</li>")
    print("</ul>")

print("<hr><p>Environment Variables:</p><pre>")
for k, v in sorted(os.environ.items()):
    print(f"{k}={v}")
print("""<a href="/">Back to Home</a>""")
print("</pre></body></html>")
